workers/unity/Packages/com.improbable.gdk.core/Worker/WorkerConnectorBase.cs

```diff
@@ -146,19 +148,19 @@ protected virtual void HandleWorkerConnectionFailure()
                 catch (ConnectionFailedException e)
                 {
                     logger.HandleLog(LogType.Error,
-                        new LogEvent($"Failed attempt to create worker")
+                        new LogEvent($"Failed attempt {remainingAttempts} to create worker")
                             .WithField("WorkerType", workerType)
                             .WithField("Message", e.Message));
-                    attempts--;
+                    remainingAttempts--;
                 }
             }
 
             throw new ConnectionFailedException(
-                $"Exceeded maximum connection attempts ({attempts})",
+                $"Tried to connect {maxAttempts} times - giving up.",
                 ConnectionErrorReason.ExceededMaximumRetries);
         }
 
-        private bool ShouldUseLocator()
+        private static bool ShouldUseLocator()
```

Comment:
We will likely have to make this virtual in the near future. Not really a big deal and we can do that when we need to though.